### PR TITLE
EES-4624 add official statistics in development release type

### PR DESCRIPTION
**Pull request by amyb-hiveit**
**Draft: waiting on the backend validation https://dfedigital.atlassian.net/browse/EES-4646.**

Adds a new release type 'Official statistics in development':
- for new releases this replaces 'Experimental statistics' which will no longer be an option when creating a release
- the release type for existing  'Experimental statistics' releases does not change and it is still an option when they are edited.

Also:
- converted `ReleaseSummaryForm` to React Hook Form and did some tidying around this
- slightly annoyingly I ended up having to create a RHF version of `FormNumberInput` because of `value={!value && value !== 0 ? '' : value}` on the original which makes the input unusable in RHF and stuff breaks without it in Formik forms.

### CODE REVIEW

File: src/explore-education-statistics-common/src/components/form/rhf/RHFFormNumberInput.tsx [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+} from '@common/components/form/FormBaseInput';
+import React from 'react';
+
+export interface RHFFormNumberInputProps extends FormBaseInputProps {
+  defaultValue?: number;
+  min?: number;
+  max?: number;
+}
+
+export default function RHFFormNumberInput(props: RHFFormNumberInputProps) {
```
Comment (ntsim) [inline src/explore-education-statistics-common/src/components/form/rhf/RHFFormNumberInput.tsx:12]: Given this'll only really get used by the form field component, probably worth just throwing this in the same file as `RHFFormFieldNumberInput` and not exporting it (same for the props)

**Review by ntsim — CHANGES_REQUESTED** [review]

File: src/explore-education-statistics-admin/src/pages/release/ReleaseCreatePage.tsx [hunk 2]
```diff
@@ -48,19 +38,22 @@ const ReleaseCreatePage = ({
 }: RouteComponentProps<MatchProps>) => {
   const { publicationId } = match.params;
 
-  const { value: model } = useAsyncRetry<Model>(async () => {
-    const [templateRelease, publication] = await Promise.all([
-      publicationService.getPublicationReleaseTemplate(publicationId),
-      publicationService.getPublication(publicationId),
-    ]);
+  const { value: model, isLoading } = useAsyncRetry<Model>(async () => {
```
[the remark this answers is not in the document]
Comment (amyb-hiveit) [reply src/explore-education-statistics-admin/src/pages/release/ReleaseCreatePage.tsx:41]: I did consider converting to to React Query but decided converting the form to RHF was enough refactoring for one PR (in the spirit of trying to keep PRs a bit smaller and quicker to review).

[the remark this answers is not in the document]
Comment (ntsim) [reply src/explore-education-statistics-admin/src/pages/release/ReleaseCreatePage.tsx:41]: Fair enough! Do you want to do a follow up PR on the heels of this one, or leave it for another time?

**Review by ntsim — APPROVED** [review]

File: src/explore-education-statistics-admin/src/pages/release/ReleaseCreatePage.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (amyb-hiveit) [reply src/explore-education-statistics-admin/src/pages/release/ReleaseCreatePage.tsx:41]: I'm going to leave it for another time as in the middle of something else now. But I'm going to have a chat to Cam about getting more of a plan for doing the incremental updates for this and RHF.

**Review by duncan-at-hiveit — APPROVED** [review]

**Pull request merged**